pkg/controller/noderesource/node_resource_controller.go

```diff
@@ -0,0 +1,117 @@
+type NodeResourceReconciler struct {
+	client.Client
+	Log      logr.Logger
+	Recorder record.EventRecorder
+}
+
+//+kubebuilder:rbac:groups=node-resource.crane.io,resources=noderesources,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=node-resource.crane.io,resources=noderesources/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=node-resource.crane.io,resources=noderesources/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the NodeResource object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *NodeResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Log.Info("node resource reconcile", "node-resource", req.NamespacedName, req.Name)
+
+	nodePre := &predictionapi.NodePrediction{}
+	// 1.get NodePrediction result
+	err := r.Client.Get(ctx, req.NamespacedName, nodePre)
+	if err != nil {
+		r.Recorder.Event(nodePre, v1.EventTypeNormal, "FailedGetNodePrediction", err.Error())
+		r.Log.Error(err, "get node prediction error", "node-prediction", req.NamespacedName, req.Name)
+		return ctrl.Result{}, err
+	}
+
+	// TODO node timeserices api resp extract
+	// 2.get current node info
+	node := &v1.Node{}
+	nodeKey := client.ObjectKey{
+		Namespace: "",
+		Name:      "",
+	}
+	if err := r.Client.Get(ctx, nodeKey, node); err != nil {
+		r.Recorder.Event(node, v1.EventTypeNormal, "FailedGetNode", err.Error())
+		r.Log.Error(err, "get node error", "node", nodeKey.Namespace, nodeKey.Name)
+		return ctrl.Result{}, err
+	}
+	nodeCopy := node.DeepCopy()
+
+	// 3.build node status
+	nextPredictionResourceStatus := &nodePre.Status
+	//resources := make(map[v1.ResourceName]resource.Quantity)
+	for key, nextPossible := range nextPredictionResourceStatus.NextPossible {
+		for _, timeSeries := range nextPossible {
+			var resourceValue int64
+			if result, err := strconv.ParseInt(timeSeries.Value, 10, 64); err != nil {
+				r.Log.Error(err, "parse extend resource value error, resource value: %s", timeSeries.Value)
+				resourceValue = result
+			}
+			// cpu memory
+			if key == string(v1.ResourceCPU) {
+				nodeCopy.Status.Capacity["ext-resource.node.crane.io/cpu"] = *resource.NewQuantity(resourceValue, resource.DecimalSI)
+			}
+
+			if key == string(v1.ResourceMemory) {
+				nodeCopy.Status.Capacity["ext-resource.node.crane.io/memory"] = *resource.NewQuantity(resourceValue, resource.DecimalSI)
+			}
+		}
+	}
+
+	// 4.update Node status extend-resource info
+	// https://kubernetes.io/zh/docs/tasks/administer-cluster/extended-resource-node/
```

Comment:
Can it be changed to https://kubernetes.io/docs/tasks/administer-cluster/extended-resource-node/ ?